src/main/java/roomescape/admin/repository/time/H2ReservationTimeRepository.java

```diff
@@ -0,0 +1,63 @@
+package roomescape.admin.repository.time;
+
+import java.sql.PreparedStatement;
+import java.sql.Time;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+import roomescape.admin.domain.ReservationTime;
+
+@Repository
+@RequiredArgsConstructor
+public class H2ReservationTimeRepository implements ReservationTimeRepository {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    @Override
+    public Long save(final ReservationTime reservationTime) {
+        String sql = "insert into reservation_times (start_at) values (?)";
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+
+        jdbcTemplate.update(connection -> {
+            PreparedStatement ps = connection.prepareStatement(sql, new String[]{"id"});
+            ps.setTime(1, Time.valueOf(reservationTime.getStartAt()));
+            return ps;
+        }, keyHolder);
+
+        return keyHolder.getKey().longValue();
+    }
+
+    @Override
+    public Optional<ReservationTime> findById(final Long id) {
+        String sql = "select * from reservation_times where id = ?";
+        List<ReservationTime> reservationTimes = jdbcTemplate.query(sql, (rs, rowNum) ->
+                        new ReservationTime(rs.getLong("id"),
+                                rs.getTime("start_at").toLocalTime()),
+                id
+        );
+
+        if (reservationTimes.size() == 1) {
```

Comment:
> 제가 생각하기에 데이터 단건 조회 시 데이터가 없는 것 또한 정상적인 조회의 결과일 수 있으므로 예외를 터뜨리는 것이 적절하지 않다고 생각이 들었습니다.
> 
> queryForObject() 를 사용하면 조회 쿼리의 결과가 1건이 아니면 예외가 발생(결과가 0개 -> EmptyResultDataAccessException, 결과가 2개 이상 -> IncorrectResultSizeDataAccessException)하므로 반드시 예외 처리를 해서 Optional.empty()를 반환하도록 해야 했습니다.
> 
> 그래서 예외를 발생시키고 catch하는 방식보다는 처음부터 예외가 아예 발생하지 않는 query() 를 사용해서 size() 검사를 하는 것이 더 안전하고, 가독성 좋고(예외 처리 로직이 사라졌으므로), 성능 측면에서도 유리하다고 생각이 들었습니다.
>
> 피케이는 어떻게 생각하시나요?

`query()` 를 사용해서 `size()`검사하는 것 좋네요 👍

`id`가 primary key라서 그럴 경우가 없을 것 같긴 하지만,
결과가 2개 이상 나왔을 경우 데이터를 놓치게 됩니다. 
"값이 있음"을 검사하는 조건문으로 변경하는 것이 더 명확할 것 같아요.

Reply:
```suggestion
        if (!reservationTimes.isEmpty()) {
```
size()==1보다는 이렇게 수정하면 좀 더 명확해질까요?

Reply:
저는 id가 unique함을 가정하고 조회가 0건, 1건인 경우만 가정했었는데요,
2개 이상의 결과가 조회된 경우는 트랜잭션의 일관성에 어긋나는 예측 밖의 결과라서 예외가 발생해야 하지 않을까 싶습니다...!

Reply:
기존의 구현에서 결과가 2개 이상 나왔을 때 데이터를 놓치게 된다는 점만 유의해서 헤일러가 판단 하시기에 더 명확한 방법으로 구현 해주시면 됩니다. :)